core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/ViewValidator.java

```diff
@@ -1,305 +0,0 @@
-/*
- * Copyright 2016-2020 Crown Copyright
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package uk.gov.gchq.gaffer.store.schema;
-
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Sets;
-import org.apache.commons.lang3.ArrayUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import uk.gov.gchq.gaffer.data.element.IdentifierType;
-import uk.gov.gchq.gaffer.data.element.function.ElementAggregator;
-import uk.gov.gchq.gaffer.data.element.function.ElementFilter;
-import uk.gov.gchq.gaffer.data.element.function.ElementTransformer;
-import uk.gov.gchq.gaffer.data.elementdefinition.view.View;
-import uk.gov.gchq.gaffer.data.elementdefinition.view.ViewElementDefinition;
-import uk.gov.gchq.gaffer.store.StoreTrait;
-import uk.gov.gchq.koryphe.ValidationResult;
-import uk.gov.gchq.koryphe.composite.Composite;
-import uk.gov.gchq.koryphe.signature.Signature;
-import uk.gov.gchq.koryphe.tuple.binaryoperator.TupleAdaptedBinaryOperator;
-import uk.gov.gchq.koryphe.tuple.function.TupleAdaptedFunction;
-import uk.gov.gchq.koryphe.tuple.predicate.TupleAdaptedPredicate;
-
-import java.util.LinkedHashSet;
-import java.util.Map;
-import java.util.Set;
-
-/**
- * An {@code ViewValidator} validates a view against a {@link Schema}
- * {@link uk.gov.gchq.gaffer.data.elementdefinition.view.ViewElementDefinition}.
- * Checks all function input and output types are compatible with the
- * properties and identifiers in the Schema and the transient properties in the
- * View.
- */
-public class ViewValidator {
```

Comment:
Again, for the sake of backwards compatibility, can we maintain this class in the codebase. We don't have to use it but if someone has extended the class to use, they would be pretty annoyed if it disappeared in a non-major release. We can put a note on it saying it'll be removed in V2 but for now, keep it in place.